src/transformers/quantizers/quantizer_torchao.py

```
@@ -137,6 +155,20 @@ def update_dtype(self, dtype):
                 dtype = torch.float32
         return dtype
 
+    def get_state_dict_and_metadata(self, model, safe_serialization: Optional[bool] = False):
+        """
+        If the model is safe serializable, we flatten the state dict of tensor subclasses so that it is compatible with
+        the safetensors format.
+        """
+        if (
+            type(self.quantization_config.quant_type) in SUPPORTED_SAFE_SERIALIZATION_CONFIGS
+            and safe_serialization
+            and version.parse(importlib.metadata.version("torchao")) >= version.parse("0.14.0")
+        ):
+            return flatten_tensor_state_dict(model.state_dict())
```

we should also raise an error here if the user try to load a safetensors file when he doesn't a sufficient version of torchao ! If we just return the state_dict, we will get a strange error.